libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -759,9 +760,11 @@ bool WaveClip::Paste(
 
    // Assume No-fail-guarantee in the remaining
    MarkChanged();
-   auto sampleTime = 1.0 / GetRate();
-   mEnvelope->PasteEnvelope
-      (s0.as_double()/mRate + GetSequenceStartTime(), newClip->mEnvelope.get(), sampleTime);
+   const auto sampleTime = 1.0 / GetRate();
+   const auto timeOffsetInEnvelope =
+      s0.as_double() * thisStretchRatio / mRate + GetSequenceStartTime();
+   mEnvelope->PasteEnvelope(
+      timeOffsetInEnvelope, newClip->mEnvelope.get(), sampleTime);
```

Comment:
I see the commit that fixes envelopes.  I read through all uses of envelopes in this file, to see if it is complete.

Do these need any changes?  I think not, but let me know the code paths were tested too and recommended to QA.

`InsertSilence` 

`AppendSilence`

`ClearAndAddCutLine`

`ExpandCutLine`

Reply:
Not tested but examined: the other places pass unstretched time values to the envelope methods, which looks correct. I added QA test cases nevertheless 👍